src/components/shared/DocToolbar/Toolbar.js

```diff
@@ -0,0 +1,82 @@
+import React, { useRef, useState } from 'react';
+import PropTypes from 'prop-types';
+import { withNamespaces } from 'react-i18next';
+import { Button, Header, Icon, Label, Menu, Popup } from 'semantic-ui-react';
+
+import ShareBtn from './ShareBtn';
+import BookmarkBtn from './BookmarkBtn';
+import CopyTextBtn from './CopyTextBtn';
+import CopyLinkBtn from './CopyLinkBtn';
+import clsx from 'clsx';
+
+const DocToolbar = ({ t, url, text, source, position, setPinned, isPinned }) => {
+  const [open, setOpen] = useState(!!url);
+  const contextRef = useRef();
+
+  const handleToggle = () => {
+    setOpen(!open);
+  };
+
+  return open && (<div
+    className={clsx('search-on-doc--bar-position', { 'pinned': isPinned })}
+    ref={contextRef}
+    style={{ top: `${position.y}px` }}
+  >
+    <div className="search-on-doc--toolbar">
+      <Popup
+        content={t('share-text.disable-share')}
+        trigger={<Button
+          inverted
+          floated={isPinned ? 'none' : 'right'}
+          icon="close"
+          circular
+          onClick={handleToggle}
+        />}
+      />
+      <Popup
+        content={isPinned ? t('share-text.unpin') : t('share-text.pin')}
+        trigger={<Button
+          inverted={!isPinned}
+          className="pin"
+          icon="thumbtack"
+          floated="right"
+          onClick={setPinned}
+        />}
+      />
+      {
+        !isPinned &&
+          (
+            <>
+              <Header
+                as="h3"
+                textAlign="center"
+              >
+                <Header.Content>
+                  {t('share-text.docbar-title')}
+                </Header.Content>
+              </Header>
+              <Menu inverted borderless>
+                <ShareBtn url={url}/>
+                <CopyLinkBtn text={url}/>
+                <CopyTextBtn text={text}/>
+                {source && <BookmarkBtn source={source} close={handleToggle}/>}
+                {/*<TagBtn />*/}
```

Comment:
What are those two? Do we need them?

Reply:
temporary hidden will used on next PR

remove here